### PR TITLE
Fix adaptive captcha tests

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/41043
Resolves https://github.com/brave/brave-browser/issues/41044

Adaptive captcha unit tests were crashing in ASAN builds on MacOS. This seemed to be caused when closing the widget, which is controlled via an animation that is kicked off by a timer. In exploring how to address this, I finally came to the conclusion that the animation was adding a lot of unwarranted complexity for no real benefit, so I decided to remove it. I also took the opportunity to clean up a few other things:

- Removed an unused `profile` argument that was being passed around to many call sites.
- Removed an unused `by_user` argument from the `Close` delegate which currently served no purpose.
- Fixed `BraveTooltipPopup::Show` to actually show the tooltip

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by zenparsing — APPROVED** [review]
Nice simplification!

File: browser/brave_ads/tooltips/ads_tooltips_controller.cc [hunk 1]
```diff
@@ -12,15 +12,11 @@
...
-#include "chrome/browser/profiles/profile.h"
 #include "components/grit/brave_components_strings.h"
 
 namespace brave_ads {
 
-AdsTooltipsController::AdsTooltipsController(Profile* profile)
-    : profile_(profile) {
-  CHECK(profile_);
-}
+AdsTooltipsController::AdsTooltipsController() {}
```
Comment (tmancey) [inline browser/brave_ads/tooltips/ads_tooltips_controller.cc:19]: nit: `= default;`

**Review by tmancey — APPROVED** [review]

File: browser/ui/views/brave_tooltips/brave_tooltips_unittest.cc [hunk 2]
```diff
@@ -105,8 +91,7 @@ TEST_F(BraveTooltipsTest, CancelButtonPressed) {
   EXPECT_CALL(mock_tooltip_delegate_, OnTooltipShow(testing::_)).Times(1);
   EXPECT_CALL(mock_tooltip_delegate_, OnTooltipCancelButtonPressed(testing::_))
       .Times(1);
-  EXPECT_CALL(mock_tooltip_delegate_, OnTooltipClose(testing::_, testing::_))
-      .Times(1);
+  EXPECT_CALL(mock_tooltip_delegate_, OnTooltipClose(testing::_)).Times(1);
```
Comment (tmancey) [inline browser/ui/views/brave_tooltips/brave_tooltips_unittest.cc:94]: nit: `Times(1)` is not needed

Comment (emerick) [reply browser/ui/views/brave_tooltips/brave_tooltips_unittest.cc]: Ahh, thank you, fixed.

**Comment by aseren** [comment]
The change looks good! FYI, chromium has pattern of disabling animation for testing: `void DisableAnimationForTesting(bool disable);`, so this maybe could be used here as well.

File: browser/brave_ads/tooltips/ads_tooltips_controller.cc (hunk 1 above)
Comment (emerick) [reply browser/brave_ads/tooltips/ads_tooltips_controller.cc]: Fixed, thanks for catching that!

**Comment by emerick** [comment]
> The change looks good! FYI, chromium has pattern of disabling animation for testing: `void DisableAnimationForTesting(bool disable);`, so this maybe could be used here as well.

I hadn't noticed that, though I did toy with doing something like that initially. I think in this particular use case, I'd just like to keep the mechanics of this notification as simple as possible. Will definitely keep that in mind for the future, though, as it's not the first time I've run into this with animations!

**Pull request merged**